srv/modules/runners/smoketests.py

```diff
@@ -0,0 +1,185 @@
+
+    def bluestore(self):
+        """
+        Return the first two devices set to bluestore
+        """
+        osds = {}
+        for device in self.devices[:2]:
+            osds[device] = {'format': 'bluestore'}
+        for device in self.devices[2:]:
+            osds[device] = {'format': 'none'}
+
+        self.base['ceph']['storage']['osds'] = osds
+        return self.base
+
+    def bluestore2(self):
+        """
+        Return the first two devices set to bluestore with a separate
+        wal and db on the third device
+        """
+        osds = {}
+        for device in self.devices[:2]:
+            osds[device] = {'format': 'bluestore',
+                            'db': self.devices[2:3][0],
+                            'db_size': '100M',
+                            'wal': self.devices[2:3][0],
+                            'wal_size': '100M'}
+        for device in self.devices[2:]:
+            osds[device] = {'format': 'none'}
+
+        self.base['ceph']['storage']['osds'] = osds
+        return self.base
+
+    def bluestore3(self):
+        """
+        Return the first two devices set to bluestore, the db on the third
+        and the wal on the fourth.
+        """
+        osds = {}
+        for device in self.devices[:2]:
+            osds[device] = {'format': 'bluestore',
+                            'db': self.devices[2:3][0],
+                            'db_size': '100M',
+                            'wal': self.devices[3:4][0],
+                            'wal_size': '100M'}
+
+        for device in self.devices[2:]:
+            osds[device] = {'format': 'none'}
+
+        self.base['ceph']['storage']['osds'] = osds
+        return self.base
+
+    def bluestored(self):
+        """
+        Return the first two devices set to encrypted bluestore
+        """
+        osds = {}
+        for device in self.devices[:2]:
+            osds[device] = {'format': 'bluestore',
+                            'encryption': 'dmcrypt'}
+        for device in self.devices[2:]:
+            osds[device] = {'format': 'none'}
+
+        self.base['ceph']['storage']['osds'] = osds
+        return self.base
+
+
+def pillar(minion, configuration):
+    """
+    Generate a pillar configuration to overwrite the existing pillar.  Removing
+    keys from the pillar is cumbersome.  Rely on overriding unnecessary disks
+    with a format of 'none'.
+    """
+    local = salt.client.LocalClient()
+    devices = local.cmd(minion, 'cephdisks.filter', [])
+    stpl = SmoketestPillar(devices[minion])
+    return stpl.create(configuration)
+
+
+def checklist(minion, configuration):
+    """
+    Save a checklist of devices on the minion
+    """
+    local = salt.client.LocalClient()
+    devices = local.cmd(minion, 'cephdisks.filter', [])
+    stpl = SmoketestPillar(devices[minion])
+    contents = stpl.checklist(configuration)
+    local.cmd(minion, 'file.write', ['/tmp/checklist', contents])
+    return ""
+
+
+def help_():
```

Comment:
Seems like there is somehting missing here?

Reply:
This is a policy question:  the smoketest runner lives in the qa package.  Which way do we want to go with help messages?  Do they follow all runners or only those in the main package?